python/kserve/test/test_grpc_server.py

```diff
@@ -0,0 +1,662 @@
+            },
+        ],
+    )
+
+    model_infer_method = server.invoke_unary_unary(
+        method_descriptor=(
+            grpc_predict_v2_pb2.DESCRIPTOR.services_by_name[
+                "GRPCInferenceService"
+            ].methods_by_name["ModelInfer"]
+        ),
+        invocation_metadata={},
+        request=request,
+        timeout=20,
+    )
+
+    response, _, code, _ = model_infer_method.termination()
+    response = await response
+    response_dict = MessageToDict(
+        response,
+        preserving_proto_field_name=True,
+    )
+    assert code == grpc.StatusCode.OK
+    assert response_dict == {
+        "model_name": "TestModel",
+        "id": "123",
+        "outputs": [
+            {
+                "name": "fp32_output",
+                "datatype": "FP32",
+                "shape": ["8"],
+                "contents": {"fp32_contents": [6.8, 2.8, 4.8, 1.4, 6.0, 3.4, 4.5, 1.6]},
+            },
+            {
+                "name": "int32_output",
+                "datatype": "INT32",
+                "shape": ["8"],
+                "contents": {"int_contents": [6, 2, 4, 1, 6, 3, 4, 1]},
+            },
+            {
+                "name": "string_output",
+                "datatype": "BYTES",
+                "shape": ["8"],
+                "contents": {
+                    "bytes_contents": [
+                        "Q2F0",
+                        "RG9n",
+                        "V29sZg==",
+                        "Q2F0",
+                        "RG9n",
+                        "V29sZg==",
+                        "RG9n",
+                        "V29sZg==",
+                    ]
+                },
+            },
+            {
+                "name": "uint8_output",
+                "datatype": "UINT8",
+                "shape": ["8"],
+                "contents": {"uint_contents": [6, 2, 4, 1, 6, 3, 4, 1]},
+            },
+            {
+                "name": "bool_input",
+                "datatype": "BOOL",
+                "shape": ["8"],
+                "contents": {
+                    "bool_contents": [
+                        True,
+                        False,
+                        True,
+                        False,
+                        True,
+                        False,
+                        True,
+                        False,
+                    ]
+                },
+            },
+        ],
+    }
+
+
+@pytest.mark.asyncio
+@patch(
+    "kserve.protocol.grpc.servicer.to_headers", return_value=[]
+)  # To avoid NotImplementedError from trailing_metadata function
+async def test_grpc_raw_inputs(mock_to_headers, server):
+    """
+    If we receive raw inputs then, the response also should be in raw output format.
+    """
+    raw_input_contents = [
+        np.array([6.8, 2.8, 4.8, 1.4, 6.0, 3.4, 4.5, 1.6], dtype=np.float32).tobytes(),
+        np.array([6, 2, 4, 1, 6, 3, 4, 1], dtype=np.int32).tobytes(),
+        serialize_byte_tensor(
```

Comment:
Wondering if this is the same as `ndarray.tobytes`
https://numpy.org/doc/stable/reference/generated/numpy.ndarray.tobytes.html